### PR TITLE
s3_bucket transfer accleration

### PR DESCRIPTION
**Pull request by abraverm**
Add support for S3 Bucket Transfer Acceleration:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/transfer-acceleration.html

### CODE REVIEW

**Comment by abraverm** [comment]
Integation test failed on "AccessDenied", I have verified the integration test with my own AWS account with S3 admin user, so I have no idea what permissions are missing or how to change them for the CI to pass.

**Comment by abraverm** [comment]
ping

**Comment by alinabuzachis** [comment]
@abraverm Thank you for taking time to work on this feature idea. Can you please open a pull request and add the missing permission `PutBucketAccelerateConfiguration` to this file https://github.com/ansible/aws-ci-admin/blob/main/aws/policy/storage-services.yaml. Thank you.

**Comment by abraverm** [comment]
I have updated the PR to match the recent changes but CI is still blocked by https://github.com/ansible/aws-ci-admin/pull/5

**Comment by gravesm** [comment]
recheck

**Review by hakbailey — COMMENTED** [review]
This looks really good! See a few minor comments inline.

**Comment by abraverm** [comment]
@hakbailey thank you for the review, I have addressed your comments in my last commit

**Comment by abraverm** [comment]
I have no clue why the CI failed, the integration tests passed locally

**Comment by abraverm** [comment]
recheck

**Comment by abraverm** [comment]
recheck

File: plugins/modules/s3_bucket.py [hunk 1]
```diff
@@ -348,6 +360,11 @@
     type: dict
     returned: I(state=present)
     sample: 'public-read'
+accelerate_enabled:
+    description: S3 bucket acceleration status.
+    type: bool
+    returned: I(state=present)
```
Comment (hakbailey) [inline plugins/modules/s3_bucket.py:366]:
```suggestion
    returned: O(state=present)
```

File: plugins/modules/s3_bucket.py [hunk 2]
```diff
@@ -832,6 +849,49 @@ def handle_bucket_object_lock(s3_client, module: AnsibleAWSModule, name: str) ->
...
+    """
+    accelerate_enabled = module.params.get("accelerate_enabled")
+    accelerate_enabled_result = False
+    accelerate_enabled_changed = False
+    try:
+        accelerate_status = get_bucket_accelerate_status(s3_client, name)
+        accelerate_enabled_result = accelerate_status
+    except is_boto3_error_code(["NotImplemented", "XNotImplemented"]) as e:
+        if accelerate_enabled is not None:
+            module.fail_json(msg="Fetching bucket transfer acceleration state is not supported")
```
Comment (hakbailey) [inline plugins/modules/s3_bucket.py:871]:
```suggestion
            module.fail_json_aws(e, msg="Fetching bucket transfer acceleration state is not supported")
```

File: plugins/modules/s3_bucket.py [hunk 3]
```diff
@@ -832,6 +849,49 @@ def handle_bucket_object_lock(s3_client, module: AnsibleAWSModule, name: str) ->
...
+    except is_boto3_error_code("AccessDenied") as e:  # pylint: disable=duplicate-except
+        if accelerate_enabled is not None:
+            module.fail_json(msg="Permission denied fetching transfer acceleration for bucket")
```
Comment (hakbailey) [inline plugins/modules/s3_bucket.py:874]:
```suggestion
            module.fail_json_aws(e, msg="Permission denied fetching transfer acceleration for bucket")
```

File: plugins/modules/s3_bucket.py [hunk 4]
```diff
@@ -832,6 +849,49 @@ def handle_bucket_object_lock(s3_client, module: AnsibleAWSModule, name: str) ->
     return object_lock_result
 
 
+def handle_bucket_accelerate(s3_client, module: AnsibleAWSModule, name: str) -> tuple[bool, dict]:
```
Comment (hakbailey) [inline plugins/modules/s3_bucket.py:852]:
```suggestion
def handle_bucket_accelerate(s3_client, module: AnsibleAWSModule, name: str) -> tuple[bool, bool]:
```

File: plugins/modules/s3_bucket.py [hunk 5]
```diff
@@ -973,6 +1038,47 @@ def create_bucket(s3_client, bucket_name: str, location: str, object_lock_enable
...
+        None
+    """
+
+    s3_client.put_bucket_accelerate_configuration(Bucket=bucket_name, AccelerateConfiguration={"Status": "Suspended"})
+
+
+@AWSRetry.exponential_backoff(max_delay=120, catch_extra_error_codes=["NoSuchBucket", "OperationAborted"])
+def get_bucket_accelerate_status(s3_client, bucket_name) -> bool:
+    """
+    Disable transfer accelerate status of the S3 bucket.
```
Comment (hakbailey) [inline plugins/modules/s3_bucket.py:1071]:
```suggestion
    Get transfer accelerate status of the S3 bucket.
```

**Review by hakbailey — COMMENTED** [review]
Just a few more minor comments. This will also need to be rebased onto main to fix the conflicts. Thanks!

**Review by alinabuzachis — COMMENTED** [review]
@abraverm Thank you for working on this feature idea. Can you also please add a changelog fragment (https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) fro this change?

**Review by gravesm — APPROVED** [review]

**Comment by abraverm** [comment]
Thank you for merging the PR and I'm sorry for the delay on my part to respond.

**Pull request merged**